scheds/rust/scx_layered/src/bpf/main.bpf.c

```diff
@@ -1843,6 +1847,38 @@ static __noinline bool match_one(struct layer_match *match,
 		// See https://github.com/sched-ext/scx/issues/610 for more details.
 		return p->tgid == p->pid && match->is_group_leader;
 	}
+	case MATCH_IS_USING_GPU: {
+			if (!enable_gpu_support)
+				return match->is_using_gpu;
+			bool pid_match = false;
+			for(int i = 0; i < MAX_GPU_PIDS; i++) {
+				if (cur_gpu_pids[i] == 0)
+					break;
+				if (cur_gpu_pids[i] == p->pid) {
```

Comment:
Why not use a hashmap?

Reply:
I *think* you have to use `BPF_MAP_TYPE_USER_RINGBUF` to write to write to a map user space side and read from a that map bpf side. 

I'm gonna try using UserRingbuf again. I couldn't find an example of using that outside of the tests for it so I was trying to avoid it.

Reply:
IIUC the list of pids is generated in user-space and the BPF part is the consumer. In this case you can definitely use a `BPF_MAP_TYPE_HASH` and you can update the items via libbpf-rs.

If you need an example of how to use `BPF_MAP_TYPE_USER_RINGBUF` you can take a look at https://github.com/sched-ext/scx/blob/dfe03fea83ba53d60884051f97d13fc2413a3f23/rust/scx_rustland_core/assets/bpf/main.bpf.c#L154, however in this case I think you can simply use a hash map.